bin/anthology/utils.py

```diff
@@ -71,6 +71,21 @@ def deconstruct_anthology_id(anthology_id):
     else:
         return (collection_id, str(int(rest[0:1])), str(int(rest[1:])))
 
+def deconstruct_anthology_volume(anthology_volume):
```

Comment:
This behaves differently from `deconstruct_anthology_id` in a very subtle way:

+ `deconstruct_anthology_id("W18-0100")` returns `("W18", "1", "0")`
+ `deconstruct_anthology_volumes("W18-01")` returns `("W18", "01")`

I think this is not intuitive. I would also advise against duplicating this code, and instead modifying `deconstruct_anthology_id` so it also works with the shorter volume IDs.

The same goes for `build_anthology_id`, which could be used to get the proper padded format for the DOIs when needed.

Reply:
Thanks for this comment.  Agree that I missed out how the original function behaves.  I will modify.  How would you like the `deconstruct_anthology_id` to function?  Just to accept either a paper or a volume identifier (assumed to be correct)?

Reply:
I'm open to other suggestions, but I think the following would be the most logical behavior:

+ `deconstruct_anthology_id("W18-0100")` returns `("W18", "1", "0")` (just as it does now)
+ `deconstruct_anthology_id("W18-01")` returns `("W18", "1", None)`

The main thing to change would be to check if `rest[1:]` or `rest[2:]` is an empty string, and using `None` there instead of the `str(int(...))` conversion.

Your calling code could then simply throw away the third tuple element if you know for certain you're passing a volume ID.

Reply:
Done in 0ab0ead .  Thanks!